Some of the steps in this simulation pile can generate large amounts of data. Particularly you should make sure that the sequences directory has enough room to store at least a few GB of data (depending on the simulation settings).

Structure:

genomics\_simulation.sh
* A master script which test the type I and II error rates of calling polymorphic sites, ...

* sequences
  * destination of all the fastq, bam, or derived files that will be created during simulation.
* sequencing\_simulation
  * the location of any scripts that generate sequence from a reference.
* assembly\_simulation
  * the location of any scripts that assemble the reference genome.
* analysis\_files
  * the locate of any temporary or final files generated by programs designed to analyze the sequence data (i.e. the output of GCTA, mapgd, etc.) 


# Reference simulation: By in large genomes do not have a random structure, but instead transpositions, duplications, selective constraint etc. create substantial sequence similarity between different regions of  the genome. This may create systematic errors when attempting to align reads to a reference genome/population, and we would like to capture these potential errors in our simulation. You may need to run 'make' in this directory before the pipeline works.
* Configurable files:
  * seed.fa
    * The initial seed sequence used in simulations.
      * mutation\_simulation2.py 
      * Specify the rates of various mutations on line 25.
* NEEDS: NOTHING
* MAKES: REFERENCE FILE
* STATUS: DONE

# Population simulation: Individuals within a population are related to each other by a pedigree structure. For instance, in finite populations we cannot model the genotypes of two individuals as being independent samples. Nor can we assume that the coalescent trees between freely recombining sites are independent of each other, because the underlying pedigree which relates individuals influences the coalescent tree. We explicitly model this pedigree. This step of the simulation of a trait value, selection on that value, geographically structured population, population expansion/contraction, and many other options. You may need to run 'make' in this directory before the pipeline works. 
	Configuration:
		This step is controled by arguments passed to the program 'population_sim'
	NEEDS: NOTHING
	MAKES: VARIANT PRESENCE/ABSENSE, PHENOTYPIC SCORES, PEDIGREE FILE, TRUE DOGE COVARIANCE 
	STATUS: DONE

# Variant simulation: Variants themselves are not random, but occur at different rates depending on the type of variant and the local sequences context. 

	NEEDS: REFERENCE, VARIANT PRESENCE/ABSENCE
	MAKES: INDIVIDUAL FASTAS
	TODO: ADD CONTEXT MODIFIRES

# Sequencing simulation: Sequencers do not just make random errors when sequencing DNA. Luckily many programs have been developed to simulate sequencing, and we can just make use of the programs that already exist.

	NEEDS: INDIVIDUAL FASTAS, REFERENCE
	MAKES: BAMFILES
	STATUS: DONE

# Analysis pipelines:
	Labeled frequency calling:
		Run simulations with the file:	
		NEEDS: BAMFILES, REFFERENCE
		STATUS: MAPGD, GATK, BCFTOOLS, ANGSD
	Relatedness:
		Run simulations with the file:	
		NEEDS: BAMFILES, REFERENCE, PHENOTYPIC SCORES
		STATUS: MAPGD, GCTA, PLINK
	Pooled frequency calling:
		Run simulations with the file:	
		NEEDS: BAMFILE, REFERENCE
		STATUS: MAPGD, BRESEQ
